app.go

```diff
@@ -911,5 +936,5 @@ func (a *Application) ReadOnlyState() State {
 		a.lastBlockHeader,
 		nil, // TODO: last block hash!
 		a.GetValidatorSet,
-	)
+	).WithEVMState(a.EvmStore.GetSnapshot(a.lastBlockHeader.Height).StateDB(a.EVMState.Database()))
```

Comment:
Should call `s.EvmStore.GetSnapshot(storeStateSnapshot.Block().Height)` to ensure both snapshots will be for the same height/version.